server/tests/plugin/auror_testssl/test_agent.py

```diff
@@ -0,0 +1,81 @@
+# This file is part of sner4 project governed by MIT license, see the LICENSE.txt file.
+"""
+auror_testssl plugin agent tests
+"""
+
+import json
+import os
+import ssl
+from uuid import uuid4
+
+import pytest
+from pytest_httpserver.httpserver import HTTPServer
+from werkzeug.serving import make_ssl_devcert
+
+from sner.agent.core import main as agent_main
+from sner.lib import file_from_zip
+
+from sner.plugin.auror_testssl.agent import AgentModule
+
+
+@pytest.fixture
+def https_server(tmpworkdir):
+    """
+    HTTPS server fixture dubbed from pytest-httpserver.
+    The original fixture does not work in conjunction with function scoped tmpworkdir fixture.
+    """
+
+    cert, key = make_ssl_devcert("testcert")
+    context = ssl.SSLContext(ssl.PROTOCOL_TLS_SERVER)
+    context.load_cert_chain(cert, key)
+
+    server = HTTPServer(ssl_context=context)
+    server.start()
+    yield server
+    server.clear()
+    if server.is_running():
+        server.stop()
+
+
+@pytest.mark.skipif("PYTEST_SLOW" not in os.environ, reason="testssl tool is slow")
```

Comment:
as auror main implementation will move to sner we need the test cases to run all the time, so we must not skip it's testing

if the test speed will be the issue, we'll speedup the test rather than skipping it (we skiped testssl tests as the module was only PoC)